### ISSUE TITLE
Module Id not included in the results of module twin queries

### ISSUE THREAD

**Issue by rajsachdev-bn** [Azure/azure-iot-sdk-java#340]
- OS and version used: Ubuntu 14.04

- Java runtime used: jdk 8u171

- SDK version used: 1.15.0

# Description of the issue:

The result of an SqlQuery across all modules in an IotHub does not include the moduleId in the modules returned.

Looks like the issue lies in the following link. The module id is never extracted from the json.
https://github.com/Azure/azure-iot-sdk-java/blob/master/service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/devicetwin/DeviceTwin.java#L486

# Code sample exhibiting the issue:

```
     //creating the sql query
      SqlQuery sqlQuery = SqlQuery.createSqlQuery("*", FromType.MODULES, null, null);
    
      //executing the sql query
      Query twinQuery = twinClient.queryTwin(sqlQuery.getQuery());
     
       //iterating through the results
      while (twinClient.hasNextDeviceTwin(twinQuery)) {
          // ISSUE: the following line does not store any moduleId into the device object.
          // Looks like the moduleId is never parsed from the module twin json.
          DeviceTwinDevice device = twinClient. getNextDeviceTwin(twinQuery);
          System.out.println(device);
      }
```
# Console log of the issue:

```
Device ID: device-1-f5e06842b
ETag: AAAAAAAAAAE=
Version: 3
Tags:{}
Reported Properties: {}
Desired Properties: {}
```

# Need Support?
* Have a feature request for SDKs? Please post it on [User Voice](https://feedback.azure.com/forums/321918-azure-iot) to help us prioritize.
* Have a technical question? Ask on [Stack Overflow](https://stackoverflow.com/questions/tagged/azure-iot-hub) with tag “azure-iot-hub”
* Need Support? Every customer with an active Azure subscription has access to support with guaranteed response time.  Consider submitting a ticket and get assistance from Microsoft support team
* Found a bug? Please help us fix it by thoroughly documenting it and filing an issue on GitHub (C, Java, .NET, Node.js, Python).

**Comment by timtay-microsoft** [comment]
You are right, this is a bug on our side. I'll look into fixing this soon. Thanks for filing this issue!

**Comment by timtay-microsoft** [comment]
The fix for this issue has been checked into master now, if you'd like to try it out.

**Comment by timtay-microsoft** [comment]
The fix for this issue has been released as of service client 1.15.1, so I'm going to close this issue now. Feel free to re-open if you still see any issues here.

**Comment by az-iot-builder-01** [comment]
@rajsachdev-bn, thank you for your contribution to our open-sourced project! Please help us improve by filling out this 2-minute [customer satisfaction survey](https://forms.office.com/Pages/ResponsePage.aspx?id=v4j5cvGGr0GRqy180BHbRx8jpG79OudPqY-BojcKjkJUMFFBQjgxNkw5ODJIRjBWTzI0Nk85R0pUQiQlQCN0PWcu)
